packages/scope-manager/tests/fixtures/type-declaration/mapped-named.ts.shot

```diff
@@ -95,21 +66,39 @@ ScopeManager {
       ],
     },
     MappedTypeScope$2 {
-      block: TSMappedType$5,
+      block: TSMappedType$4,
       isStrict: true,
       references: [
-        Reference$2,
-        Reference$3,
-        Reference$4,
+        Reference$2 {
+          identifier: Identifier<"k">,
+          isRead: true,
+          isTypeReference: true,
+          isValueReference: false,
+          isWrite: false,
+          resolved: null,
+        },
+        Reference$3 {
+          identifier: Identifier<"k">,
+          isRead: true,
+          isTypeReference: true,
+          isValueReference: false,
+          isWrite: false,
+          resolved: null,
+        },
+        Reference$4 {
+          identifier: Identifier<"k">,
+          isRead: true,
+          isTypeReference: true,
+          isValueReference: false,
+          isWrite: false,
+          resolved: null,
+        },
+        Reference$5,
       ],
-      set: Map {
-        "k" => Variable$4,
-      },
```

Comment:
this is a problem - this shouldn't have disappeared!
the type variable `k` is still defined in this scope.

Reply:
Aha! Nice spot, thanks.